lib/pure/parseopt2.nim

```diff
@@ -0,0 +1,133 @@
+## argument parser (akin to POSIX getopt)
+## 
+## Can handle:
+## -c c_val
+## -c:c_val
+## -c=c_val
+## -abcc_val
+## -abc c_val
+## -abc:c_val
+## -abc=c_val
+## --long val
+## --long:val
+## --long=val
+## --long=    (empty val)
+## -- -positional1 --positional2
+## 
+## -    is treated as positional
+## --   skipped once; everything after this is treated as positional
+## ---  is treated as long arg "-"
+## 
+## -:   undefined behavior (separator after dash)
+
+import options, strutils
+
+type
+  OptKind* = enum
```

Comment:
`./foo --whatever ''`  is valid, so it might be worth differentiating long and short with and without values?

Presently the end programmer can't differentiate the result of an `Opt` with an optional value receiving and empty value or none at all.

An alternative approach would be a flag or the ability to query position (how much has been consumed) so the information can be reconstructed.

Reply:
The user should know, because `shortWithVal` and `longWithVal`.

Reply:
Apologies if I'm being slow, but given an long/short with an optional argument, putting it into `shortWithVal` or `longWithVal` would result in an exception. That would then have to be recovered just to handle an optional value. :/

Reply:
the library doesn't have support for optional argument

Reply:
We have a bunch in the compiler code base, basically any `on|off` flags treat no value as default "on", it'd be a shame to lose that.

Reply:
this is added in latest commits